opentelemetry-resourcedetector-gcp/src/opentelemetry/resourcedetector/gcp_resource_detector/_gke.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from dataclasses import dataclass
+
+from opentelemetry.resourcedetector.gcp_resource_detector import (
+    _gce,
+    _metadata,
+)
+
+# TODO: remove when Python 3.7 is dropped
+from typing_extensions import Literal
+
+KUBERNETES_SERVICE_HOST_ENV = "KUBERNETES_SERVICE_HOST"
+
+
+def on_gke() -> bool:
```

Comment:
Should this also check for the presence of the metadata server?

Reply:
The metadata server is assumed to be available at this point in the detector.  See https://github.com/GoogleCloudPlatform/opentelemetry-operations-python/pull/233/files#diff-690ac73bc4cd4aba0fbc7ff5f2233fb14ec3c86c407f07b6b39b22dc7718aa45R53-R54.

Reply:
I'm OK to sprinkle the check throughout the `on_x()` methods since it's cached anyway.

Fwiw, I'm not planning to expose these functions (remember we thought the GCP Logging client libs might use these but things changed).

Reply:
Ah, I missed the not metadata check (and I had looked for it).